docs/data/data-grid/column-recipes/ColumnSizingPersistWidthOrder.js

```diff
@@ -0,0 +1,87 @@
+import * as React from 'react';
+import {
+  DataGridPro,
+  useGridApiRef,
+  gridColumnFieldsSelector,
+} from '@mui/x-data-grid-pro';
+import Button from '@mui/material/Button';
+
+const rows = [
+  {
+    id: 1,
+    username: '@MUI',
+    age: 20,
+  },
+];
+
+const useColumnsState = (apiRef, columns) => {
+  const [widths, setWidths] = React.useState({});
+  const [orderedFields, setOrderedFields] = React.useState(
+    columns.map((column) => column.field),
+  );
```

Comment:
It should at least be `useState(() => ...)`, the `.map` operation can be expensive.